wagtail/contrib/modeladmin/helpers/url.py

```diff
@@ -6,16 +6,26 @@
 
 
 class AdminURLHelper:
-    def __init__(self, model):
+    def __init__(self, model, base_url_path=None):
         self.model = model
         self.opts = model._meta
+        self.base_url_path = base_url_path
 
```

Comment:
What if we initialised the page URL path with the current default behaviour (if the class attribute is blank). 

Or maybe its logic is inside a get method in the class?

Reply:
That makes sense. I understand what you meant.

Just another idea, but would creating a set_base_url_path method a reasonable solution? Instead of passing base_url_path into the constructor, I will create a set_base_url_path that will be called in options.py's constructor - thus, eliminating the needs for having a warning if the overrided url helper class does not have base_url_path parameter. So, something like

```
        self.url_helper = self.get_url_helper_class()(self.model)
        if self.base_url_path:
                self.url_helper.set_base_url_path(self.base_url_path)
```

And the code in AdminURLHelper would be something like

```
class AdminURLHelper:
    def __init__(self, model, base_url_path=None):
        self.model = model
        self.opts = model._meta
        self.set_base_url_path(None)
    
    def set_base_url_path(self, base_url_path):
        if base_url_path:
              # set modified path
              self.base_url_path = ....
         else:     
              # set default path
```

If you feel like this is not appropriate, I would do a get_base_url_path just like you suggest. Just let me know your opinion!

Reply:
not a big fan of mutating the instance where possible as it introduces other ways for things to break .

an alternative approach might be to pass the class' ModelAdmin to the url helper class, this is similar to your current approach but will allow for more nuanced behaviour if needed in the future.

Reply:
I have made some of the changes - I decide to follow my current approach, but if you have any comments, please let me know

Reply:
yeah, I agree, better to avoid adding the whole machine unless we need it and better to be explicit.

I think this looks and feels much better.